### DISCUSSION TITLE
Micronaut Grpc with Jaeger

### DISCUSSION THREAD

**Discussion by lucaspaixao-dev** [micronaut-projects/micronaut-core#7546] [Q&A]
Hello,

I'm creating a backend service using micronaut and grpc, and I want to integrate with jaeger. Looking on the internet, I have not found any site to help me. The official documentation doesn't have so many examples.
Can anyone help me?
Thanks!

**Comment by shingjo** [comment]
In addition to, [User Documentation on Tracing](https://micronaut-projects.github.io/micronaut-tracing/latest/guide/#jaeger) there is a [Tracing with Jaeger guide](https://guides.micronaut.io/latest/micronaut-microservices-distributed-tracing-jaeger.html).

Quick explanation:

The Micronaut framework provides three annotations to create or continue an existing span and include additional information to the span.

**@NewSpan** on a method creates a new span.
- `@NewSpan("name")`  creates a span with the given name.
- `@NewSpan` creates a span with the method name.

**@ContinueSpan** on a method continues the existing span, primarily used in conjunction with `@SpanTag`. 

**@SpanTag** on method parameter/argument to include the value to the span, the method must be annotated with either `@NewSpan` or `@ContinueSpan`.
- `@SpanTag("tag.name")` adds a tag/label to the span with the given name.
- `@SpanTag` adds a tag to the span with the parameter name.

Note: `@NewSpan` and `@ContinueSpan` can be used on non-private methods.
